pkg/cli/config/config.go

```diff
@@ -351,3 +352,56 @@ func getRawKubeConfigLoader(kubeconfigPath string) clientcmd.ClientConfig {
 	// Return the clientcmd.ClientConfig (equivalent to ToRawKubeConfigLoader)
 	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
 }
+
+// ServerConnectionFlagsChanged returns true if any server connection flags are set.
+func ServerConnectionFlagsChanged(cmd *cobra.Command) bool {
+	return cmd.Flags().Changed("host") ||
+		cmd.Flags().Changed("token") ||
+		cmd.Flags().Changed("insecure-skip-tls-verify") ||
+		cmd.Flags().Changed("api-path")
+}
+
+// BuildDirectClientConfig builds a client.Config from CLI flags (host, token, api-path, insecure-skip-tls-verify).
+func BuildDirectClientConfig(p common.Params) (*client.Config, error) {
+	host := p.Host()
+	token := p.Token()
+	if host == "" || token == "" {
+		return nil, errors.New("--host and --token flag must be set if using direct connection flags")
+	}
+	rc := &rest.Config{
+		Host: host,
+	}
+	if p.APIPath() != "" {
+		rc.APIPath = p.APIPath()
+	}
+	if p.Token() != "" {
+		rc.BearerToken = token
+	}
+
+	rc.Insecure = p.SkipTLSVerify()
+	// Optionally set timeout (default 30s)
+	rc.Timeout = 30 * time.Second
+
+	rc.APIPath = path.Join(rc.APIPath, Path)
+
+	rc.GroupVersion = &schema.GroupVersion{
+		Group:   Group,
+		Version: Version,
+	}
+	u, pth, err := rest.DefaultServerUrlFor(rc)
```

Comment:
note: that feels strange, the function returns 2 values (and an error), then we update the first value with the second one `u.Path = pth`. DefaultServerUrlFor needs some redesign IMO.

Reply:
@enarha I am just using the `DefaultServerUrlFor` from the `client-go` library.